openedx/core/djangoapps/content/learning_sequences/api/processors/special_exams.py

```diff
@@ -54,26 +57,49 @@ def exam_data(self, pruned_course_outline: UserCourseOutlineData) -> SpecialExam
                         continue
 
                     special_exam_attempt_context = None
-                    try:
-                        # Calls into edx_proctoring subsystem to get relevant special exam information.
-                        # This will return None, if (user, course_id, content_id) is not applicable.
-                        special_exam_attempt_context = get_attempt_status_summary(
-                            self.user.id,
-                            str(self.course_key),
-                            str(sequence.usage_key)
-                        )
-                    except ProctoredExamNotFoundException:
-                        log.info(
-                            'No exam found for {sequence_key} in {course_key}'.format(
-                                sequence_key=sequence.usage_key,
-                                course_key=self.course_key
+
+                    # if exam waffle flag enabled, then use exams logic
+                    if exams_ida_enabled(self.course_key):
+                        # add short description based on exam type
+                        if sequence.exam.is_practice_exam:
+                            exam_type = gettext_noop('Practice Exam')
+                        elif sequence.exam.is_proctored_enabled:
+                            exam_type = gettext_noop('Proctored Exam')
+                        elif sequence.exam.is_time_limited:
+                            exam_type = gettext_noop('Timed Exam')
+                        # todo: Do I need this default?
+                        else:
+                            exam_type = gettext_noop('Exam')
+
+                        summary = {}
+                        summary.update({
+                            'short_description': _(exam_type),
+                            'suggested_icon': 'fa-pencil-square-o'
+                        })
+
+                        special_exam_attempt_context = summary
+                        # todo: what about due date? Do I need to include this from edx-when?
```

Comment:
[MST-1528](https://2u-internal.atlassian.net/browse/MST-1528) mentions the due date but it seems like this may not be necessary or is being pulled from somewhere else? The due date is not in the summary when calling into edx proctoring. Let me know if I should include this.

Reply:
I don't think any new code is needed since the due date is already returned as part of this request. The UI already knows about it and just jams the description an due date together here: https://github.com/openedx/frontend-app-learning/blob/fd94da0a432ebd9df561c8ea9b49afe880c38a6e/src/course-home/outline-tab/SequenceLink.jsx#L78